### PR TITLE
[SIMPLE_FORMS] feat: pull out submission building logic from archive builder service object

### PR DESCRIPTION
**Pull request by pennja**
## Summary

- This work updates the archive builder to not be so closely tied to VFF form submissions
- This work also adds RSpec unit test coverage

## Related issue(s)

- [Create PoC for adding filled in PDFs to an S3 bucket](https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1633)

## Testing done

- [x] *New code is covered by unit tests*

## Requested Feedback

Any

### CODE REVIEW

File: modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb [hunk 1]
```diff
@@ -37,73 +32,34 @@ def run
...
+      end
+
       def process_submission_files
         write_pdf
         write_as_json_archive if include_json_archive
         write_as_text_archive if include_text_archive
         write_attachments unless attachments.empty?
         write_manifest if include_manifest
         write_metadata
+      rescue => e
```
Comment (Thrillberg) [inline modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb:57]: On line 54, if we're initializing `attachments` as `nil` then calling `empty?` on `nil` will error out.

**Review by Thrillberg — COMMENTED** [review]
Looks good to me except for one little thing I think I found. LMK if it's a real concern!

File: modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb (hunk 1 above)
Comment (pennja) [reply modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb:57]: Good call, I went back and forth on this with hydrate_submission_data but I forgot to also account for the post-submission route.

Comment (pennja) [reply modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb:57]: Fixed!

**Review by Thrillberg — APPROVED** [review]
Looks good. Thanks!!

**Review by ryan-mcneil — APPROVED** [review]
👍

**Comment by ryan-mcneil** [comment]
@pennja  I'm going to have to merge it because of the weird CI error. please ping me if/when you want me to do so

**Comment by pennja** [comment]
> @pennja I'm going to have to merge it because of the weird CI error. please ping me if/when you want me to do so

@ryan-mcneil Thanks for the approval. Feel free to approve!!

**Pull request merged**